src/Neo/Network/P2P/Payloads/Signer.cs

```diff
@@ -66,6 +66,33 @@ public class Signer : IInteroperable, ISerializable
             /*AllowedGroups*/       (Scopes.HasFlag(WitnessScope.CustomGroups) ? AllowedGroups.GetVarSize() : 0) +
             /*Rules*/               (Scopes.HasFlag(WitnessScope.WitnessRules) ? Rules.GetVarSize() : 0);
 
+        public bool Equals(Signer other)
+        {
+            if (ReferenceEquals(this, other))
+                return true;
+            return Account == other.Account &&
+                Scopes == other.Scopes &&
+                AllowedContracts.SequenceEqual(other.AllowedContracts) &&
+                AllowedGroups.SequenceEqual(other.AllowedGroups) &&
+                Rules.SequenceEqual(other.Rules);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
```

Comment:
== null already done in is sentence

Reply:
`==` checks for `null` and sends to this method.